lxd/storage_volumes_utils.go

```diff
@@ -86,16 +87,6 @@ func storagePoolVolumeUpdateUsers(s *state.State, projectName string, oldPoolNam
 
 // storagePoolVolumeUsedByGet returns a list of URL resources that use the volume.
 func storagePoolVolumeUsedByGet(s *state.State, requestProjectName string, vol *db.StorageVolume) ([]string, error) {
-	// Handle instance volumes.
-	if vol.Type == cluster.StoragePoolVolumeTypeNameContainer || vol.Type == cluster.StoragePoolVolumeTypeNameVM {
-		volName, snapName, isSnap := api.GetParentAndSnapshotName(vol.Name)
-		if isSnap {
-			return []string{api.NewURL().Path(version.APIVersion, "instances", volName, "snapshots", snapName).Project(vol.Project).String()}, nil
-		}
-
-		return []string{api.NewURL().Path(version.APIVersion, "instances", volName).Project(vol.Project).String()}, nil
-	}
-
```

Comment:
For containers, doesn't it make sense to return early?

You are claiming it doesn't affect performance, but how can that be if we are running `VolumeUsedByInstanceDevices` and `VolumeUsedByProfileDevices` unnecessarily for containers?

I may have missed something here, if so sorry about this.

Reply:
Maybe I wasn't super clear. You're right, the way I've done this definitely affects performance for containers unnecessarily.

I'm still hoping to get this feature implemented for containers too, so this shouldn't be unnecessary for long.

Reply:
Hmm I see. Probably best to keep this efficient so we don't run the risk of making a release/round of backports that only include this commit and not the future implementation for containers. What do you think?

Reply:
I re-introduced the check for containers, I'll remove it when I implement container root attachments.